packages/browser/src/utils/task-queue.ts

```diff
@@ -0,0 +1,210 @@
+/**
```

Comment:
I don't exactly like how we need to use this class, but it works, so please just merge if you think this is good.

IMO the way we should consume this is `new AsyncProcessor(iterator, { timeBudgetMs: 30 }).processEach(item => doSomething(item))`

This is definitely different from your idea where each item in the array is a function and you're just calling those functions, BUT it does align more with how you're using this class (list of items that need to be processed iwth the same function everytime). It basically means to me you're trying to use the wrong abstraction for the job, hence why you had to add those 2 helper functions.